wallet/src/main/scala/org/bitcoins/wallet/internal/RescanHandling.scala

```diff
@@ -107,10 +107,9 @@ private[wallet] trait RescanHandling extends WalletLogger {
         _ = logger.info(
           s"Beginning to search for matches between ${startHeight}:${endHeight} against ${scripts.length} spks")
         range = startHeight.to(endHeight)
-        matched <- FutureUtil.batchExecute(
+        matched <- FutureUtil.batchAndSyncExecute(
```

Comment:
We should really add a unit test for this, it should be trivial no? Basically make the rescan bigger then `batchSize`?

Reply:
Added